compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala

```diff
@@ -401,26 +400,16 @@ final class Conversions(
       .build
   }
 
-  def convertPartialTransaction(ptx: SPartialTransaction): proto.PartialTransaction = {
-    val builder = proto.PartialTransaction.newBuilder
-      .addAllNodes(ptx.nodes.map(convertNode).asJava)
-      .addAllRoots(
-        ptx.context.children.toImmArray.toSeq.sortBy(_.index).map(convertTxNodeId).asJava
-      )
+  def convertPartialTransaction(incomplete: IncompleteTransaction): proto.PartialTransaction = {
+    val tx = incomplete.transaction
 
-    @tailrec
-    def unwindToExercise(
-        contextInfo: SPartialTransaction.ContextInfo
-    ): Option[SPartialTransaction.ExercisesContextInfo] = contextInfo match {
-      case ctx: SPartialTransaction.ExercisesContextInfo => Some(ctx)
-      case ctx: SPartialTransaction.TryContextInfo =>
-        unwindToExercise(ctx.parent.info)
-      case _: SPartialTransaction.RootContextInfo => None
-    }
+    val builder = proto.PartialTransaction.newBuilder
+      .addAllNodes(tx.nodes.map(convertNode).asJava)
+      .addAllRoots(tx.roots.toList.map(convertTxNodeId).asJava)
```

Comment:
`incomplete.transaction` use only the node until the last failing exercise, it should from the root.